kernel/src/hil/hw_debug.rs

```diff
@@ -0,0 +1,31 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! Interfaces for interacting with debug hardware integrated in various SoCs.
+//! Currently allows reading the cycle counter.
+
+pub trait CycleCounter {
+    /// Enable and start the cycle counter.
+    fn start(&self);
+
+    /// Stop the cycle counter.
+    /// Does nothing if the cycle counter is not present.
+    fn stop(&self);
+
+    /// Return the current value of the cycle counter.
+    fn count(&self) -> u32;
```

Comment:
The RISC-V counters are 64-bit, so this will have to support a `u64` as well

Reply:
Good call, done